### PR TITLE
 [ENH] Make FCN TensorFlow layers dynamic via n_layers, filter_sizes, and kernel_sizes

### PR DESCRIPTION
**Pull request by varun-kht**
#### Reference Issues/PRs
 Refers #9340 

#### What does this implement/fix? Explain your changes.
 Makes the number of convolutional layers, filter sizes, and kernel sizes in the FCN TensorFlow implementation dynamic by adding `n_layers`, `filter_sizes`, and `kernel_sizes` parameters to `FCNNetwork`, `FCNClassifier`, and `FCNRegressor`. Previously the network was hardcoded to 3 Conv1D layers with filters [128, 256, 128] and kernel sizes [8, 5, 3]. Both filter_sizes and kernel_sizes accept a list or tuple, with validation at __init__ level.

#### Does your contribution introduce a new dependency? If yes, which one?
 no

#### What should a reviewer concentrate their feedback on?
-The dynamic loop in sktime/networks/fcn.py build_network method
-Validation logic for filter_sizes and kernel_sizes in FCNNetwork.__init__
-Backward compatibility — default behaviour is fully preserved

#### Did you add any tests for the change?
 Updated `get_test_params` in `FCNNetwork`, `FCNClassifier`, and `FCNRegressor` to test different values of `n_layers`, `filter_sizes`, and `kernel_sizes`, including both list and tuple inputs.

#### Any other comments?
took reference from #9506  and #9607 
##### For all contributions
- [x] I've added myself to the [list of contributors] 
 - [ ] Optionally, for added estimators: I've added myself and possibly to the `maintainers` tag - do this if you want to become the owner or maintainer of an estimator you added.
  - [x] The PR title starts with either [ENH], [MNT], [DOC], or [BUG]. [BUG] - bugfix, [MNT] - CI, test framework, [ENH] - adding or improving code, [DOC] - writing or improving documentation or docstrings.

##### For new estimators
- [ ] I've added the estimator to the API reference  
- [ ] I've added one or more illustrative usage examples to the docstring
- [ ] If the estimator relies on a soft dependency, I've set the `python_dependencies` tag and ensured
  dependency isolation,

### CODE REVIEW

File: sktime/classification/deep_learning/fcn.py [hunk 1]
```diff
@@ -19,11 +19,8 @@ class FCNClassifier(BaseDeepClassifier):
 
     Parameters
     ----------
-    should inherited fields be listed here?
     n_epochs : int, default = 2000
         the number of epochs to train the model
-    callbacks : list of tf.keras.callbacks.Callback objects, default=None
-        List of callbacks to apply during training.
```
Comment (Faakhir30) [inline sktime/classification/deep_learning/fcn.py:26]: why are you removing `callbacks` from docstring?
Was this generated by AI? If yes, then please do mention in PR that AI was used and always have a look at changes generated.

File: sktime/classification/deep_learning/fcn.py [hunk 2]
```diff
@@ -48,6 +45,15 @@ class FCNClassifier(BaseDeepClassifier):
...
         specify the optimizer and the learning rate to be used.
+    n_layers : int , default = 3
+        number of convolutional layers in the network.
+    filter_sizes : list or tuple of int , default = [128,256,128]
+        number of filters for each convolutional layer.
+        must have length equal to n_layers.
+    kernel_sizes : list or tuple of int , default = [8,5,3]
+        kernel size for each convolutional layer.
+        must have length equal to n_layers.
+
```
Comment (Faakhir30) [inline sktime/classification/deep_learning/fcn.py]: I think we can avoid introducing `n_layers` param in this one, we can use `len(filter_sizes)` to determine number of convolution layers.
Reason why I did not suggested similar approach in your other PR #9607 is because MLPNetwork was already supporting `float` value or `tuple` for `dropout` param. So its not possible to predict number of layers if a `float` value is passed. 
A workaround could have been implemented by depricating `float` type from `dropout` specifically, but that would have been complex, so proceeded with adding explicit `n_layers` to control layers count.

File: sktime/networks/fcn.py [hunk 3]
```diff
@@ -45,11 +53,42 @@ def __init__(
...
+            raise ValueError(
+                f"filter_sizes must have length equal to n_layers={self.n_layers} ,"
+                f"but got length {len(self.filter_sizes)}."
+            )
+        # Kernel_sizes length check
+        if len(self.kernel_sizes) != self.n_layers:
+            raise ValueError(
+                f"kernel_sizes must have length equal to n_layers={self.n_layers}, "
+                f"but got length {len(self.kernel_sizes)}."
+            )
```
Comment (Faakhir30) [inline sktime/networks/fcn.py]: These checks can then be simplified to following:
```suggestion

        if len(self.filter_sizes) != len(self.kernel_sizes):
            raise ValueError(
                f"Length of `filter_sizes` {len(self.filter_sizes)} must match "
                f"the number of convolutional layers determined by the length of tuple "
                f"`kernel_sizes` {len(self.kernel_sizes)}."
            )
```

File: sktime/regression/deep_learning/fcn.py [hunk 4]
```diff
@@ -45,6 +44,14 @@ class FCNRegressor(BaseDeepRegressor):
         whether the layer uses a bias vector.
     optimizer : keras.optimizers object, default = Adam(lr=0.01)
         specify the optimizer and the learning rate to be used.
+    n_layers : int , default =3
+        number of convolutional layers in the network.
```
Comment (Faakhir30) [inline sktime/regression/deep_learning/fcn.py]: Same comment regarding `n_layers`.

File: sktime/classification/deep_learning/fcn.py (hunk 1 above)
Comment (varun-kht) [reply sktime/classification/deep_learning/fcn.py:26]: hi @RecreationalMath @Faakhir30 it is not ai generated ,sometimes i do take some suggestions but here i didn't use it. and the reason why i am  removing `callbacks` from the docstring is beacuse it is repeating itself just after `batch_sizes` in docstring(i.e line no. 26) .please do review it.
thank you!

File: sktime/networks/fcn.py (hunk 3 above)
Comment (varun-kht) [reply sktime/networks/fcn.py]: hi @RecreationalMath @Faakhir30 do you want to also want to remove the type check for `filter_sizes ` and `kernel_sizes`(i.e from 65-79 line no.) beacuse i think they might be a good defensive measure ,won't it ?

File: sktime/classification/deep_learning/fcn.py (hunk 2 above)
Comment (varun-kht) [reply sktime/classification/deep_learning/fcn.py]: sure will be accounting that in next push.
Thank you !

File: sktime/classification/deep_learning/fcn.py (hunk 1 above)
Comment (Faakhir30) [reply sktime/classification/deep_learning/fcn.py:26]: Oh cool, I didnt notice that.

No worries, I do take suggestions too, we only have to ensure that we understand the changes like you are aware of reasoning here, thanks.

File: sktime/networks/fcn.py (hunk 3 above)
Comment (Faakhir30) [reply sktime/networks/fcn.py]: Other PRs linked to this issue are not having explicit type checks for `filter_sizes` etc, so skipped in above suggestion.
Sure, you can keep type checks.

Comment (varun-kht) [reply sktime/networks/fcn.py]: Okay ,sure would be updating stuff with next  push .
Thank you !

**Comment by varun-kht** [comment]
hi @RecreationalMath @Faakhir30  have updated pr accounting suggestions. would love feedback.
Thank you!

File: sktime/classification/deep_learning/fcn.py [hunk 5]
```diff
@@ -89,6 +93,8 @@ def __init__(
         activation_hidden="relu",
         use_bias=True,
         optimizer=None,
+        filter_sizes=None,
+        kernel_sizes=None,
```
Comment (Faakhir30) [inline sktime/classification/deep_learning/fcn.py]: I do see `None` being overriden at network layer for these `filter_sizes`, `kernel_sizes` defaults. But it would be cleaner to have those defualts set in parameter list instead in tuples `kernel_sizes=(8,5,3)` and `filter_sizes=(128,256,128)`

Right now if you do:
```python
est = FCNClassifier()
print(est.filter_sizes)
```
this will print `None`, while expected defualt to be shown is that tuple.

Suggesting tuple for default values type is intentional as tuples are immutable.

Same for other two files, set `kernel_sizes=(8,5,3)` and `filter_sizes=(128,256,128)` at parameter definations.

Comment (varun-kht) [reply sktime/classification/deep_learning/fcn.py]: @Faakhir30 sure earlier it was depending on `n_layers` so kept `None` but now since it is not dependable removing seems good idea . Will be accounting that in next push.  
Thank you!

**Comment by varun-kht** [comment]
hi @RecreationalMath @Faakhir30  have updated the pr accounting all the suggestions .would love feedback .
Thank you !

**Comment by varun-kht** [comment]
hey, @fkiraly i believe the issue has been resolved and i have cross verified with local test run .so please review for final merge. Thank you !
